src/test/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -0,0 +1,233 @@
+                                                                             .map(enhanceWithQualifiedSchema)
+                                                                             .map(toQualifiedJSONSchemaValidator)
+                                                                             .collect(Collectors.toList());
+
+        return new QualifiedJSONSchemaValidationChain(qualifiedValidators, defaultSchemaValidator);
+    }
+
+    public static int findElement(final JSONArray array, final String element) {
+
+        for (int i = 0; i < array.length(); i++) {
+            if (element.equals(array.getString(i))) {
+                return i;
+
+            }
+        }
+
+        return -1;
+    }
+}
+
+class SchemaValidationOverrides {
+    private List<OverrideDefinition> overrides;
+
+    public List<OverrideDefinition> getOverrides() {
+        return overrides;
+    }
+
+    public void setOverrides(final List<OverrideDefinition> overrides) {
+        this.overrides = overrides;
+    }
+}
+
+class Qualifier {
+    private String field;
+    private String match;
+
+    public String getField() {
+        return field;
+    }
+
+    public void setField(final String field) {
+        this.field = field;
+    }
+
+    public String getMatch() {
+        return match;
+    }
+
+    public void setMatch(final String match) {
+        this.match = match;
+    }
+}
+
+class OverrideDefinition {
+    private Qualifier qualifier;
+    private List<String> ignoredProperties;
+    private JSONObject effectiveSchema;
+
+    public Qualifier getQualifier() {
+        return qualifier;
+    }
+
+    public void setQualifier(final Qualifier qualifier) {
+        this.qualifier = qualifier;
+    }
+
+    public List<String> getIgnoredProperties() {
+        return ignoredProperties;
+    }
+
+    public void setIgnoredProperties(final List<String> ignore) {
+        this.ignoredProperties = ignore;
+    }
+
+    public JSONObject getEffectiveSchema() {
+        return effectiveSchema;
+    }
+
+    public void setEffectiveSchema(final JSONObject effectiveSchema) {
+        this.effectiveSchema = effectiveSchema;
+    }
+}
+
+class JSONSchemaValidator implements EventValidator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JSONSchemaValidator.class);
+
+    private final Schema schema;
+    private final JSONObject effectiveSchema;
+
+    public JSONSchemaValidator(final JSONObject effectiveSchema) {
+        this.effectiveSchema = effectiveSchema;
+        schema = SchemaLoader.load(effectiveSchema);
+    }
+
+    @Override
+    public boolean accepts(final JSONObject event) {
+        try {
+            schema.validate(event);
+            return true;
+        } catch (final ValidationException e) {
+            e.printStackTrace();
+            LOG.error("Validation failed: {}. Schema was: {}",
+                new Object[] {Joiner.on(";").join(e.getCausingExceptions()), effectiveSchema, e});
```

Comment:
Don't worry now about this exception handling/logging at this time. It is fully bogus.